examples/cli/src/main/java/io/unitycatalog/cli/TableCli.java

```diff
@@ -98,44 +98,70 @@ private static String createTable(TablesApi apiClient, JSONObject json)
+        throw new CliException("Storage location is not allowed for managed tables");
+      }
+      // Create staging table if format is delta
+      if (!DataSourceFormat.DELTA.name().equals(format.toUpperCase())) {
+        throw new CliException("Only delta tables are supported for managed tables: " + format);
+      }
+      CreateStagingTable createStagingTable =
+          new CreateStagingTable()
+              .catalogName(json.getString(CliParams.CATALOG_NAME.getServerParam()))
+              .schemaName(json.getString(CliParams.SCHEMA_NAME.getServerParam()))
+              .name(json.getString(CliParams.NAME.getServerParam()));
+      StagingTableInfo stagingTableInfo = tablesApi.createStagingTable(createStagingTable);
+      String stagingTableId = stagingTableInfo.getId();
+      String stagingLocation = stagingTableInfo.getStagingLocation();
+      if (stagingTableId == null || stagingLocation == null) {
+        throw new CliException("Failed to create staging table");
+      }
+      createTable.setStorageLocation(stagingLocation);
+      temporaryCredentials =
+          temporaryCredentialsApi.generateTemporaryTableCredentials(
+              new GenerateTemporaryTableCredential()
+                  .tableId(stagingTableId)
+                  .operation(TableOperation.READ_WRITE));
     } else {
-      return Paths.get(path);
+      throw new CliException("Unknown table type: " + createTable.getTableType());
     }
-  }
 
-  private static void handleTableStorageLocation(
-      String storageLocation, List<ColumnInfo> columnInfos) {
-    if (!(storageLocation.startsWith("s3://")
-        || storageLocation.startsWith("file:/")
-        || storageLocation.startsWith("/"))) {
-      throw new CliException(
-          "Storage location must start with s3:// or file:/ or be an absolute local filesystem path.");
-    }
-    if (!storageLocation.startsWith("s3://")) {
-      // local filesystem path
-      Path path = getLocalPath(storageLocation);
-      // try and initialize the directory and initiate delta log at the location
-      try {
-        DeltaKernelUtils.createDeltaTable(path.toUri().toString(), columnInfos, null);
-      } catch (Exception e) {
-        if (e.getCause() instanceof TableAlreadyExistsException) {
-          // TODO confirm the schema of the existing table matches the schema of the new table
-        } else {
-          throw new CliException("Failed to create delta table at " + path, e);
-        }
+    // try and initialize the directory and initiate delta log at the location
+    try {
+      DeltaKernelUtils.createDeltaTable(
+          createTable.getStorageLocation(), columnInfoList, temporaryCredentials);
+    } catch (Exception e) {
+      if (e instanceof TableAlreadyExistsException
+          || e.getCause() instanceof TableAlreadyExistsException) {
+        // TODO confirm the schema of the existing table matches the schema of the new table
```

Comment:
why do we need to do this? are we building `create` semantics or `create or replace` semantics?